ssp

```diff
@@ -1913,28 +1911,20 @@ sub check_for_clustering {
     chdir $cluster_dir or return;
 
     for my $dirent (@dir_contents) {
+        my ( $cluster_member, $cluster_member_hostname, $cluster_member_role );
+        my %cluster_conf;
 
         # only active cluster members have -dnsrole files
         if ( $dirent =~ m{ \A (.+)-dnsrole \z }xms ) {
             $cluster_member = $1;
 
-            if ( open my $file_fh, '<', "${cluster_member}.cache" ) {
-                my $cache_ref;
-                eval 'local $SIG{__DIE__}; local $SIG{__WARN__}; $cache_ref = Storable::fd_retrieve($file_fh);';    # from upcp.static
+            if ( open my $file_fh, '<', $cluster_member ) {
+                local $/;
+                %cluster_conf = map { ( split( /=/, $_, 2 ) )[ 0, 1 ] } split( /\n/, readline($file_fh) );
                 close $file_fh;
-                if ($cache_ref) {
-                    $cluster_member_hostname = $cache_ref->{'host'};
-                }
-                close $file_fh;
-            }
-            else {
-                $cluster_member_hostname = '?';
-            }
-
-            if ( !$cluster_member_hostname ) {
-                $cluster_member_hostname = '?';
             }
-            if ( ${cluster_member} =~ m{ \A (vps\.net|softlayer) \z}xmsi ) {
+            $cluster_member_hostname = defined $cluster_conf{host} ? $cluster_conf{host} : '?';
```

Comment:
Curious of this ternary.  either { bFqPExC7N2yggP } or { '?' }

What is the value of '?' at this point?